server/src/main/java/org/apache/cassandra/sidecar/job/OperationsJobTracker.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.job;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Tracks and stores the results of long-running jobs running on the sidecar
+ */
+public class OperationsJobTracker extends LinkedHashMap<UUID, OperationsJob>
```

Comment:
The class can use some simplifications.

Reply:
+1 we don't need a full blown map here. Also, I think internally we should keep track of Futures

Reply:
@yifan-c  Can you elaborate on simplifications?

Regarding the need for map, we already had a discussion in this thread about this - the LRU eviction policy and thread-safety is driving this need. https://github.com/apache/cassandra-sidecar/pull/139#discussion_r1830081175. 

>  I think internally we should keep track of Futures

Not sure I understand, there are no Futures referenced in this class. Can you elaborate?